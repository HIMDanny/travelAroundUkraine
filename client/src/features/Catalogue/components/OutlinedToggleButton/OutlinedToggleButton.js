import React, { useState } from 'react';
import PropTypes from 'prop-types';
import { styled, alpha, ToggleButton } from '@mui/material';

const StyledToggleButton = styled(ToggleButton)(({ theme }) => ({
    borderRadius: 40,
    border: '1px solid #EDEDED',
    background: 'none',
    fontWeight: 500,
    fontSize: 12,
    textTransform: 'none',
    padding: theme.spacing(1),
    textAlign: 'center',
    width: '100%',
    color: theme.palette.text.primary,
  
    '&:hover': {
      boxShadow: `${alpha(theme.palette.primary.main, 0.15)} 0 0 0 0.12rem`,
      borderColor: theme.palette.primary.main,
      backgroundColor: `${alpha(theme.palette.primary.main, 0.05)}`,
    },
  
    '&.Mui-selected': {
      backgroundColor: theme.palette.primary.main,
      color: theme.palette.primary.contrastText,
  
      '&:hover': {
        backgroundColor: `${alpha(theme.palette.primary.main, 0.85)}`,
      },
    },
  }));
  

  const OutlinedToggleButton = ({ value, children }) => {
    const [selected, setSelected] = useState(false);
  
    return (
      <StyledToggleButton
        value={value}
        selected={selected}
        onChange={() => setSelected(prev => !prev)}
      >
        {children}
      </StyledToggleButton>
    );
  };

  OutlinedToggleButton.propTypes = {
    children: PropTypes.any,
    value: PropTypes.number,
  };
  
  OutlinedToggleButton.defaultProps = {
    children: '',
    value: 0,
  };

  export default OutlinedToggleButton;
