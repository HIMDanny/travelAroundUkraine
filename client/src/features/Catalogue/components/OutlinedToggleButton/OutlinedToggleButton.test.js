import { render } from '@testing-library/react';
import OutlinedToggleButton from './OutlinedToggleButton';

jest.mock('@mui/material/ToggleButton', () => ({ children }) => <button>{children}</button>);

describe('OutlinedToggleButton snapshot test', () => {
  test('should OutlinedToggleButton match snapshot', () => {
    const { asFragment } = render(<OutlinedToggleButton />);
    expect(asFragment()).toMatchSnapshot();
  });
});
