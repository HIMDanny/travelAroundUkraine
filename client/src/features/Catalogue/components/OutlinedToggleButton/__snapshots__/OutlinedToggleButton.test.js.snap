// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`OutlinedToggleButton snapshot test should OutlinedToggleButton match snapshot 1`] = `
<DocumentFragment>
  <button />
</DocumentFragment>
`;
